pkg/engine/mutate/mutation.go

```diff
@@ -101,28 +102,50 @@ type patchesJSON6902Handler struct {
 	logger          logr.Logger
 }
 
-func newPatchesJSON6902Handler(ruleName string, mutate *kyverno.Mutation, patchedResource unstructured.Unstructured, logger logr.Logger) Handler {
+func newPatchesJSON6902Handler(ruleName string, mutate *kyverno.Mutation, patchedResource unstructured.Unstructured, context context.EvalInterface, logger logr.Logger) Handler {
 	return patchesJSON6902Handler{
 		ruleName:        ruleName,
 		mutation:        mutate,
 		patchedResource: patchedResource,
+		evalCtx:         context,
 		logger:          logger,
 	}
 }
 
 func (h patchesJSON6902Handler) Handle() (resp response.RuleResponse, patchedResource unstructured.Unstructured) {
-	resp.Name = h.ruleName
-	resp.Type = utils.Mutation.String()
+	ruleResponse := response.RuleResponse{
+		Name:    h.ruleName,
+		Success: false,
+		Type:    utils.Mutation.String(),
+	}
 
 	patchesJSON6902, err := convertPatchesToJSON(h.mutation.PatchesJSON6902)
 	if err != nil {
-		resp.Success = false
 		h.logger.Error(err, "error in type conversion")
-		resp.Message = err.Error()
-		return resp, h.patchedResource
+		ruleResponse.Message = err.Error()
+		return ruleResponse, h.patchedResource
+	}
+
+	var patch interface{}
+	err = json.Unmarshal(patchesJSON6902, &patch)
+	if err != nil {
+		ruleResponse.Message = err.Error()
+		return ruleResponse, h.patchedResource
+	}
+
+	patchedPatch, err := variables.SubstituteAll(h.logger, h.evalCtx, patch)
```

Comment:
As Jim mentioned [here](https://github.com/kyverno/kyverno/issues/1521#issuecomment-772053374), do we want to allow variables to be specified in `path`? What if it resolves to a list?

Reply:
I think this situation is OK, because in this case path won't apply. The more places we can use to substitute vars, the more flexible we are. Invalid `path` with an list inside must be validated further when patch is applied.

Reply:
Yes I agree we want to make it as flexible as possible, but supporting variable substitution in `path` could potentially bring unexpected results for different types, thus lead to confusing situations. Do you see any use case there?

Reply:
I see no use case for more complex types than string in `path`, but I also think that this is not the actual place where this check should be applied. We could check that the `path` is a string when we actually use this `path` and fail in other cases.